### PR TITLE
Blog app: add API endpoints

### PR DESCRIPTION
**Pull request by Alexr16**
### In this PR 🤖:
📌 Created an API endpoint to list all posts for a user.
Screen Shot 2022-07-21 at 18 06 45

📌 Created an API endpoint to list all comments for a user's post.
Screen Shot 2022-07-21 at 18 07 03

📌 Created an API endpoint to add a comment to a post. Remember that the owner of the comment is the user that makes it; it should **not** be set by passing a parameter.
Screen Shot 2022-07-21 at 18 11 28

📌 Added authentication to the API endpoints.
Screen Shot 2022-07-21 at 18 20 28

📌 API endpoints receive JSON and respond JSON as well.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,4 +1,14 @@
...
+    namespace :v1, defaults: { format: :json } do #to specify the default format
+      resources :users do
+        resources :posts, only: [:index] do
+          resources :comments, only: [:index, :create]
+        end
+      end
+    end
+  end
+
   devise_for :users
```
Comment (brytebee) [inline config/routes.rb:15]: - [x] **Required:** You should also provide a public route for sign-in/log-in just as you have done for sign-out. Without a login route, one cannot log in to get the tokens for authentication. I therefore could not verify the endpoints as authentication must be in place for them to function.

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, Team👋,

Good job so far coming so far with the progress you've made! 💪💪

## Highlights
You have done amazing in the following:
- Included a class for authentication.✔️
- Has structured routes and controllers into their respective namespace.✔️
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb (hunk 1 above)
Comment (Alexr16) [reply config/routes.rb:15]: @brytebee, thanks for your review.
The login method to get the tokens for authentication has been added.

Screen Shot 2022-07-21 at 18 49 24
.

**Review by BenMukebo — APPROVED** [review]
### Hi @Alexr16, @CRClothier, 👋 

# STATUS: Approved  🎉 

Great work on making the changes requested by a previous reviewer :1st_place_medal:  
 Your project is complete!  There is nothing else to say other than... it's time to merge it
Congratulations! 🎉  🎉 🎉 




## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @BenMukebo ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**